apps/examples/index.coffee

```diff
@@ -0,0 +1,65 @@
+express = require 'express'
+Promise = require 'bluebird-q'
+
+graphQL = require '../../lib/graphql'
+examples = require './examples.coffee'
+{ first, map, each, find } = require 'underscore'
+
+app = module.exports = express()
+
+app.set 'views', "#{__dirname}/templates"
+app.set 'view engine', 'jade'
+
+exampleChannelQuery = """
+  query example($ids: [ID]!) {
+    channels(ids: $ids) {
+      __typename
+      title(truncate: 50)
+      updated_at(relative: true)
+      user {
+        name
+      }
+      id
+      href
+      visibility
+      counts {
+        blocks
+      }
+      blocks(per: 6, sort_by: POSITION, direction: DESC, type: BLOCK) {
+        ... blockThumb
+      }
+    }
+  }
+  #{require '../../components/block_v2/queries/block.coffee'}
+"""
+
+app.get '/api/examples/:id', (req, res, next) ->
+  example = find(examples, (example) -> example.id is req.params.id)
+  send =
+    query: exampleChannelQuery
+    user: req.user
+    variables:
+      ids: example.channel_ids
+
+  graphQL(send).then ({ channels }) ->
+    example.channels = channels
+    res.json example
+
+app.get '/examples', (req, res, next) ->
+  Promise.all(
+    map first(examples, 6), (example) ->
+      send =
+        query: exampleChannelQuery
+        user: req.user
+        variables:
+          ids: first(example.channel_ids, 2)
+    
+      graphQL(send)
+  )
+  .then (results) ->
+    each results, (result, i) ->
+      examples[i].channels = result.channels
+   
+    res.locals.examples = examples
+    res.render 'index'
+  .catch(next)
```

Comment:
Not saying you have to but you might want to consider not writing CoffeeScript where we don't have to; like routes for instance. You can use the Apollo middleware's client to make queries: https://github.com/aredotna/ervell/pull/1320/files#diff-9e5e66872800e1139ae9aab6f79b83b0R36

Reply:
Ah k, will do.